### DISCUSSION TITLE
How to use grayscale image as intensity of given color?

### DISCUSSION THREAD

**Discussion by ktyl** [ImageMagick/ImageMagick#2240] [Help] [answered]
I'm trying to merge the separations created from the PDF document back into a CMYK image. I'm dealing just fine with standard CMYK separatinos by simply combining them, but I can't walk over the situation when there are additional spot colors channels. I have read through many threads related to this problem, but none of them allowed me to get the result I'm trying to achieve.

I know how to combine the images, but I'm stuck on generating the properly colorized separation plate for the spot color.

I have uploaded the example files here: https://drive.google.com/file/d/1l_aEfXFsJKLXymM-adbq5EKX_viOCxg0/view?usp=sharing
Let's say that I want to use the spot color cmyk(100%,30%,30%,10%).

I'm looking for a way to convert the input image (channel.tif) into the output image (output.tif) in a following way:
white in channel.tif -----> white in output.tif
black in channel.tif -----> full intensity of my spot color in output.tif (cmyk(100%,30%,30%,10%))

All the colors between black and white in channel.tif should give the appropriate intensity of my spot color in the output image.

Thanks in advance for any help that could guide me towards the solution.

**Comment by snibgo** [comment, marked as answer]
Here is a sample gray image (0=black and QuantumRange at white), Windows BAT syntax:
```
%IMG7%magick ^
  toes.png ^
  -colorspace Gray ^
  -auto-level ^
  channel.tiff
```

We spread these values across 4 channels, and declare them to be CMYK. Finally, we use "+level-colors".
```
%IMG7%magick ^
  -debug command ^
  channel.tiff -duplicate 3 -combine +channel -set colorspace CMYK ^
  -alpha off ^
  +level-colors cmyk(100%%,30%%,30%%,10%%),cmyk(0,0,0,0) ^
  g2.tiff
```
g2.tiff is white where channel.tiff was white, and cmyk(100%%,30%%,30%%,10%%) where channel.tiff was black.

**Reply by ktyl** [reply]
Thank you for your reply @snibgo, it helped me a lot! In my case I needed to force change the colorspace of channel.tif to gray, I'm not sure why, because it was grayscale already. I ended up with the following syntax:
```
%IMG7%magick convert ^
  channel.tiff
  -colorspace Gray ^
  -duplicate 3 ^
  -combine ^
  +channel ^
  -set colorspace CMYK ^
  -alpha off ^
  +level-colors cmyk(100%%,30%%,30%%,10%%),cmyk(0,0,0,0) ^
out.tiff
```
